src/app/clusters/general_diagnostics_server/general_diagnostics_server.cpp

```diff
@@ -219,6 +251,37 @@ class GeneralDiagnosticsDelegate : public DeviceLayer::ConnectivityManagerDelega
         ChipLogProgress(Zcl, "GeneralDiagnosticsDelegate: OnHardwareFaultsDetected");
 
         ReportAttributeOnAllEndpoints(GeneralDiagnostics::Attributes::ActiveRadioFaults::Id);
+
+        for (uint16_t index = 0; index < emberAfEndpointCount(); index++)
+        {
+            if (emberAfEndpointIndexIsEnabled(index))
+            {
+                EndpointId endpointId = emberAfEndpointFromIndex(index);
+
+                if (emberAfContainsServer(endpointId, GeneralDiagnostics::Id))
+                {
+                    // If General Diagnostics cluster is implemented on this endpoint
+                    MatterReportingAttributeChangeCallback(endpointId, GeneralDiagnostics::Id,
+                                                           GeneralDiagnostics::Attributes::ActiveRadioFaults::Id);
+
+                    // Record RadioFault event
+                    EventNumber eventNumber;
+                    EventOptions eventOptions(System::SystemClock().GetMonotonicTimestamp(), EventOptions::Type::kUrgent);
```

Comment:
Is this urgent event? Just wanna double check on it.

Reply:
I think we should mark event with priority CRITICAL as Urgent, right? The spec does not mandate it